src/pycell.rs

```diff
@@ -828,6 +829,26 @@ impl<T: PyClass> IntoPy<PyObject> for &'_ PyRef<'_, T> {
     }
 }
 
+impl<'py, T: PyClass> IntoPyObject<'py> for PyRef<'py, T> {
+    type Target = T;
+    type Output = Bound<'py, T>;
+    type Error = Infallible;
+
+    fn into_pyobject(self, _py: Python<'py>) -> Result<Self::Output, Self::Error> {
+        Ok(self.inner.clone())
+    }
+}
```

Comment:
I wonder if somehow we can manually decrement the borrow count and then just return `self.inner` to avoid the clone here (same for the `PyRefMut` case).

Reply:
Yeah, these are suboptimal... I haven't figured out a way to move the `inner` because of the `Drop` impl. But with the complete remove of the gil-refs now, maybe we can also now store a `Borrowed` in `PyRef`/`PyRefMut`, which should solve that issue I think.

Reply:
Reflecting on `Borrowed` in `PyRef` / `PyRefMut`, it creates the downside we already saw with types like `Vec<&'a str>` where they can't easily be extracted. (Because a `PyRef<'a, 'py, T>` containing a borrowed won't be able to be stored by iterating a list, for example.)

But I think on balance that might actually be a good thing, because it helps prevent locking objects to read/write for longer than desired. (The migration solution would be encouraging users to extract `Vec<Bound<'py, T>>` and then `.borrow()` for just the needed span of code.)